### DISCUSSION TITLE
Set YouTube's default color according to system-theme to avoid flashes

### DISCUSSION THREAD

**Discussion by MulverineX** [ReVanced/revanced-discussions#173] [Question]
-

**Comment by oSumAtrIX** [comment]
Which app?

**Comment by MulverineX** [comment]
Oh woops, YouTube. YTM doesn't have a light theme so it doesn't have this issue. Not currently suggesting theming for YTM.

**Comment by oSumAtrIX** [comment]
I still don't understand your issue. YouTube should automatically choose the theme of your system.

**Comment by MulverineX** [comment]
Yes the actual theme of the app, but very frequently when multitasking it will take longer to load the app even to the splash screen, and the placeholder color is white regardless of system theme.

It's similar to how on desktop browsers on potato PCs the tab will flash white before rendering the dark themed website.

This does not happen with other apps.

**Comment by oSumAtrIX** [comment]
How about setting the theme to dark?

**Comment by MulverineX** [comment]
That may work, this is an intermittent issue so I'll try it. However, it would be nice if the Manager could patch the app regardless. I haven't had to change the theme setting manually on any other Google app that follows system theme.

**Comment by oSumAtrIX** [comment]
This will never get any patch since the fix is to simply set the theme yourself in the case that fixes the issue.

**Comment by MulverineX** [comment]
Just experienced the issue again after manually setting to dark theme.

Android 12, Google Pixel 5a. YouTube app hadn't been opened in several hours & was probably not in memory. Pressed a YouTube link in Discord, white page flashed for 1< second before loading the app. I do not have any forwarding apps etc, the default app *is* YouTube.

**Comment by oSumAtrIX** [comment]
I can not replicate

**Comment by MulverineX** [comment]
Its an intermittent issue ¯\_(ツ)_/¯. Surely there are ways to search the code for the hardcoded white?

**Comment by rkn45** [comment]
Cannot replicate either, both Vanced and a regular YouTube app launch for me without "white flashes". Even after being completely cleared out of memory.

**Comment by MulverineX** [comment]
I am using stock YouTube version 17.15.33.

And again, this is a rare/intermittent issue, it isn't something you can easily reproduce.

**Comment by rkn45** [comment]
I have the same stock YT version and Vanced 17.03.38, and none of them have white splash screen on launch while using system-wide dark mode. The contrary happens when I have the light mode on, both apps have white splashes on launch. 

Using MIUI 12.5.2 on Android 11.

**Comment by rkn45** [comment]
Thinking it might be somewhat "fixable" by replacing the resources for light mode to be the same as dark mode. Perhaps it could be made into a separate optional patch.

**Comment by KrazyDeinos** [comment]
Any update on this. I have this issue sometimes when I've been using another app pretty heavy.

**Comment by oSumAtrIX** [comment]
This is a ROM issue. An app can't do much about the short flash at the beginning.

**Reply by MulverineX** [reply]
This doesn't happen with other apps and a potential solution was proposed

**Comment by oSumAtrIX** [comment]
I can reproduce with other apps.

**Comment by MulverineX** [comment]
So the tags on this are incorrect?

**Reply by oSumAtrIX** [reply]
I don't know what you mean by tags.

**Reply by MulverineX** [reply]
@oSumAtrIX 
![image](https://user-images.githubusercontent.com/12068027/224748841-8aa67dbd-6340-43e6-9e75-9fe1b54d3ea8.png)

**Reply by oSumAtrIX** [reply]
I am unable to reproduce it on YouTube but on other apps, I can remove the label, but this discussion will get stale.
